e2e/security/security.go

```diff
@@ -0,0 +1,292 @@
+		},
+		{
+			name:          "Set_gid",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"id", "-g", "|", "grep", "99"},
+			opts:          e2e.ExecOpts{Security: []string{"gid:99"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		// seccomp from json file
+		{
+			name:          "SecComp_BlackList",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"mkdir", "/tmp/foo"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          1,
+			expectSuccess: false,
+		},
+		{
+			name:          "SecComp_true",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"true"},
+			opts:          e2e.ExecOpts{Security: []string{"seccomp:./testdata/seccomp-profile.json"}},
+			exit:          0,
+			expectSuccess: true,
+		},
+		// capabilities
+		{
+			name:          "capabilities_keep",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{KeepPrivs: true},
+			exit:          0,
+			expectSuccess: true,
+		},
+		{
+			name:          "capabilities_drop",
+			image:         imagePath,
+			action:        "exec",
+			argv:          []string{"ping", "-c", "1", "8.8.8.8"},
+			opts:          e2e.ExecOpts{DropCaps: "CAP_NET_RAW"},
+			exit:          1,
+			expectSuccess: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run("priv "+tt.name, test.WithPrivilege(func(t *testing.T) {
+			_, stderr, exitCode, err := e2e.ImageExec(t, testenv.CmdPath, tt.action, tt.opts, tt.image, tt.argv)
+			if tt.expectSuccess && (exitCode != 0) {
+				t.Log(stderr, err, exitCode)
+				t.Fatalf("unexpected failure running '%v': %v", strings.Join(tt.argv, " "), err)
+			} else if !tt.expectSuccess && (exitCode != 1) {
+				t.Log(stderr, err, exitCode)
+				t.Fatalf("unexpected success running '%v'", strings.Join(tt.argv, " "))
+			}
+		}))
+	}
+}
+
+// testSecurityConfOwnership tests checks on config files ownerships
+func testSecurityConfOwnership(t *testing.T) {
+	configFile := buildcfg.SYSCONFDIR + "/singularity/singularity.conf"
+	// Change file ownership (do not try this at home)
+	err := os.Chown(configFile, 1001, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// try to run
+	t.Run("non_root_config", test.WithoutPrivilege(func(t *testing.T) {
+		_, stderr, exitCode, err := e2e.ImageExec(t, testenv.CmdPath, "exec", e2e.ExecOpts{}, imagePath, []string{"/bin/true"})
+		if exitCode != 1 {
+			t.Log(stderr, err)
+			t.Fatalf("unexpected success running /bin/true")
+		}
+	}))
+
+	// return file ownership to normal
+	err = os.Chown(configFile, 0, 0)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func testSecurity(t *testing.T) {
+	test.EnsurePrivilege(t)
+	opts := e2e.BuildOpts{
+		Force:   true,
+		Sandbox: false,
+	}
+	if b, err := e2e.ImageBuild(testenv.CmdPath, opts, imagePath, "../../examples/busybox/Singularity"); err == nil {
```

Comment:
could this use the image provided by the e2e framework? You can make sure it exists by calling e2e.EnsureImage(t)

Reply:
What I mean before is that when you call e2e.EnsuareImage() you are making sure that a specific test image exists.

Can you use that image instead of building another one?

If yes, please delete the call to e2e.ImageBuild.

If no, please add a comment as to why a specific image is required.